runtime/interpreter/interpreter_expression.go

```diff
@@ -222,6 +238,41 @@ func (interpreter *Interpreter) memberExpressionGetterSetter(memberExpression *a
 	}
 }
 
+// getReferenceValue Returns a reference to a given value.
+// Reference to an optional should return an optional reference.
+// This has to be done recursively for nested optionals.
+// e.g.1: Given type T, this method returns &T.
+// e.g.2: Given T?, this returns (&T)?
+func (interpreter *Interpreter) getReferenceValue(value Value, semaType sema.Type) Value {
```

Comment:
Does this match the checker implementation?

Reply:
Yes; checker's counter-part of this: https://github.com/onflow/cadence/blob/c16e5682fa679469cd52d9331d1f117d1275eed9/runtime/sema/check_member_expression.go#L95-L108